### PR TITLE
Release GVL for get{login,pwnam,pwuid,grnam}{,_r} calls in process.c

### PR DESCRIPTION
**Pull request by jeremyevans**
Related to GVL-releasing work in dir.c for bug 20587.

### CODE REVIEW

File: process.c [hunk 1]
```diff
@@ -5728,60 +5755,47 @@ check_gid_switch(void)
...
             rb_str_resize(maybe_result, 0);
             rb_syserr_fail(gle, "getlogin_r");
         }
 
-        rb_str_modify_expand(maybe_result, loginsize);
-        login = RSTRING_PTR(maybe_result);
-        loginsize = rb_str_capacity(maybe_result);
+        rb_str_modify_expand(maybe_result, args.namelen);
+        args.name = RSTRING_PTR(maybe_result);
+        args.namelen = rb_str_capacity(maybe_result);
```
Comment (nobu) [inline process.c:5786]: Probably this should be better to cast to `getlogin_r_size_t`.
This is an implicit down-cast, or different signedness.

File: process.c [hunk 2]
```diff
@@ -5728,60 +5755,47 @@ check_gid_switch(void)
...
         loginsize = GETLOGIN_R_SIZE_DEFAULT;
 
     VALUE maybe_result = rb_str_buf_new(loginsize);
-
-    login = RSTRING_PTR(maybe_result);
-    loginsize = rb_str_capacity(maybe_result);
-    rb_str_set_len(maybe_result, loginsize);
+    struct getlogin_r_args args;
+    args.name = RSTRING_PTR(maybe_result);
+    args.namelen = rb_str_capacity(maybe_result);
```
Comment (nobu) [inline process.c:5767]: Ditto.

File: process.c [hunk 3]
```diff
@@ -5842,25 +5881,25 @@ rb_getpwdirnam_for_login(VALUE login_name)
         }
 
         rb_str_modify_expand(getpwnm_tmp, bufsizenm);
-        bufnm = RSTRING_PTR(getpwnm_tmp);
-        bufsizenm = rb_str_capacity(getpwnm_tmp);
+        args.buf = RSTRING_PTR(getpwnm_tmp);
+        args.bufsize = rb_str_capacity(getpwnm_tmp);
```
Comment (nobu) [inline process.c:5885]: `long` to `size_t` assignment, as well.

Comment (jeremyevans) [reply process.c]: Found this affected additional cases, so I added casts for those calls as well.

**Review by nobu — APPROVED** [review]

**Comment by jeremyevans** [comment]
@nobu If you have time, can you please review https://github.com/ruby/ruby/pull/11202/commits/ca81d6f183452f59563342b3108b2a8ae5d10a72 and let me know if I got all of the casts correct?

**Pull request merged**